prov/verbs/src/windows/verbs_nd_rdma.c

```diff
@@ -185,11 +237,21 @@ int rdma_destroy_id(struct rdma_cm_id *id)
 		refcnt = id_nd->connector->lpVtbl->Release(id_nd->connector);
 		assert(!refcnt);
 	}
+
 	if (id->qp) {
 		ibv_destroy_qp(id->qp);
 		id->qp = NULL;
 	}
 	// Should also destroy any associated SRQ here.
+
+	if (id_nd->ov_file != INVALID_HANDLE_VALUE) {
+		ret = CloseHandle(id_nd->ov_file);
+		assert(ret);
+	}
+
+	ofi_mutex_destroy(&id_nd->connect_event.base.lock);
+	ofi_mutex_destroy(&id_nd->peer_event.base.lock);
+	ofi_mutex_destroy(&id_nd->listen_event.base.lock);
```

Comment:
This function is getting long.  Please break up the functionality into multiple calls.